installation.md

```diff
@@ -0,0 +1,99 @@
+
+Launchers should use the value of the `versionJson` field to correlate libraries defined in a version JSON file and a library in MODIP format.
+
+### `runInstaller`
+
+The file with an installer relation value should be executed. Note that this field is very vague and it's recommended to use something with more information such as `versionJsonInstall`.
+
+### `instanceInstall`
+
+This is a modpack/instance and should be treated as such. Required dependencies should be installed to a directory specific to this instance, then files should be extracted from a file with `"rel" "compressedOverrides"` and moved to the directory.
+
+If this is a modpack, mods should be listed as a required dependency if it is possible to do so.
+
+Files in the `compressedOverrides` zip should be in the root. For example:
+`overrides.zip/overrides/overridden_file.txt` shouldn't be used. Instead, `overrides.zip/overridden_file.txt` is better.
+
+### `other`
+
+An unknown installation method.
+
+`TODO: what other methods should be added?`
+
+---
+
+## Specifying Paths
+
+Some fields, such as `placeInDirectory`'s `directory` field, requires a path to be supplied. This path by default is relative to the root of the Minecraft Instance that this project is being installed to. If the Project is not able to be installed to a Minecraft Instance for whatever reason, it is up to launchers to supply a default relative path.
+
+Paths MUST use forward slashes to indicate nesting. Backslashes (even on Windows systems) CANNOT be used.
+
+Paths MAY start with `./`, however it is not needed. For example, `./mods` is a valid directory path, however `mods` also works and does not have the unnecessary `./`.
+
+Absolute paths MAY be specified by prefixing the path with an absolute path prefix. This prefix has two valid options:
+
+- `/` character for Unix-like systems
+- Windows device prefix path (e.g. `C:/`)
+
+Absolute prefixes are HIGHLY DISCOURAGED and should only be used when absolutely necessary. For example, a mod that installs itself to the `mods` folder SHOULD NOT use an absolute path, and instead should specify `mods`.
+
+The following are rules on paths and filenames:
+
+- Paths that contain Windows restricted filenames and characters are not allowed
+  - Restricted characters are `<`, `>`, `:`, `"`, `|`, `?`, `*`
+- Empty strings are not allowed
+  - If you need to specify the current relative path, without using a subdirectory, use `.`
+- Filenames CANNOT contain a slash or backslash
+- Filenames that end in `.` CANNOT be used
+- Filenames that end in a space CANNOT be used
+- New line and carriage return characters CANNOT be used
+- Most Unicode characters are allowed, but using normal ASCII characters (A-Z, a-z, 0-9) is recommended
+
+  - Restricted Unicode characters include NUL and CR
```

Comment:
LF? Unicode spaces? Byte order marks? Unpaired surrogates?

Note that Windows uses UTF-16 paths and Linux uses arbitrary byte strings. The sensible intersection is ASCII excluding Windows's reserved characters.